### ISSUE TITLE
TLS negotiation failed

### ISSUE THREAD

**Issue by eislambey** [amphp/http-client#255]
I'm trying to connect https://n11.com or https://hepsiburada.com then i get this error.

**Error**
```
PHP Fatal error:  Uncaught Amp\Socket\TlsException: TLS negotiation failed: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:14094460:SSL routines:ssl3_read_bytes:reason(1120) in /home/islambey/Projects/amp-http-test/vendor/amphp/socket/src/Internal/functions.php:128
Stack trace:
#0 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/NativeDriver.php(192): Amp\Socket\Internal\{closure}()
#1 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/NativeDriver.php(97): Amp\Loop\NativeDriver->selectStreams()
#2 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/Driver.php(134): Amp\Loop\NativeDriver->dispatch()
#3 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/Driver.php(72): Amp\Loop\Driver->tick()
#4 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop.php(84): Amp\Loop\Driver->run()
#5 /home/islambey/Projects/amp-http-test/get.php(20): Amp\Loop::run()
#6 {main}

Next Amp\Http\Client\SocketException: Connection to 'n11.com in /home/islambey/Projects/amp-http-test/vendor/amphp/http-client/src/Connection/DefaultConnectionFactory.php on line 156

Fatal error: Uncaught Amp\Socket\TlsException: TLS negotiation failed: stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:14094460:SSL routines:ssl3_read_bytes:reason(1120) in /home/islambey/Projects/amp-http-test/vendor/amphp/socket/src/Internal/functions.php:128
Stack trace:
#0 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/NativeDriver.php(192): Amp\Socket\Internal\{closure}()
#1 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/NativeDriver.php(97): Amp\Loop\NativeDriver->selectStreams()
#2 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/Driver.php(134): Amp\Loop\NativeDriver->dispatch()
#3 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop/Driver.php(72): Amp\Loop\Driver->tick()
#4 /home/islambey/Projects/amp-http-test/vendor/amphp/amp/lib/Loop.php(84): Amp\Loop\Driver->run()
#5 /home/islambey/Projects/amp-http-test/get.php(20): Amp\Loop::run()
#6 {main}

Next Amp\Http\Client\SocketException: Connection to 'n11.com in /home/islambey/Projects/amp-http-test/vendor/amphp/http-client/src/Connection/DefaultConnectionFactory.php on line 156

```

**Versions**
`PHP 7.4.2` `amphp/http-client 4.1.0`

`OpenSSL 1.1.1d  10 Sep 2019` `OpenSSL version number 269488207`

**Code example**
```php
<?php

use Amp\Http\Client\HttpClientBuilder;
use Amp\Http\Client\Request;
use Amp\Http\Client\Response;
use Amp\Loop;

require __DIR__ . "/vendor/autoload.php";

Loop::run(function () {
    $httpClient = HttpClientBuilder::buildDefault();

    $request = new Request("https://n11.com");
    
    /** @var Response $response */
    $response = yield $httpClient->request($request);

    echo $response->getStatus(), "\n";

});
```

**Comment by kelunik** [comment]
The server doesn't seem to like ALPN. If you set the HTTP versions to only `1.1` on the request, the client will now skip using ALPN, so the connection works.

```php
$request = new Request("https://n11.com");
$request->setProtocolVersions(['1.1']);
```

I've released v4.2.0 with the fix.

**Comment by kelunik** [comment]
@eislambey Thanks for the report, the actual fix is in https://github.com/amphp/socket/releases/tag/v1.1.1, so no restriction to HTTP/1.1 needed.
